registry/storage/garbagecollect.go

```diff
@@ -65,6 +66,11 @@ func MarkAndSweep(ctx context.Context, storageDriver driver.StorageDriver, regis
 			return nil
 		})
 
+		err = manifestEnumerator.LayersEnumerate(ctx, repoName, func(dgst digest.Digest, linkpath string) error {
```

Comment:
All you need to do here is find the layers that aren't referenced in the repository and remove. It should look similar to the global blob collection, but can be local to a repository.